### PR TITLE
JDO-844: Add SampleReadQuery test methods with variables of type String

### PR DESCRIPTION
**Pull request by mboapache**

### CODE REVIEW

File: tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo [hunk 1]
```diff
@@ -131,6 +131,9 @@ has application identity.
             <field name="phoneNumbers" persistence-modifier="persistent">
                 <map key-type="String" value-type="String"/>
             </field>
+            <field name="languages" persistence-modifier="persistent">
+                <collection type="String"/>
```
Comment (tzaeschke) [inline tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo:135]: Shouldn't that be `<collection element-type="String"/>`?

File: tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo [hunk 2]
```diff
@@ -275,6 +278,9 @@ has application identity.
             <property name="phoneNumbers" persistence-modifier="persistent">
                 <map key-type="String" value-type="String"/>
             </property>
+            <property name="languages" persistence-modifier="persistent">
+                <collection type="String"/>
```
Comment (tzaeschke) [inline tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo:282]: Shouldn't that be `<collection element-type="String"/>`?

File: tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo [hunk 3]
```diff
@@ -106,6 +106,9 @@ has datastore identity.
             <field name="phoneNumbers" persistence-modifier="persistent">
                 <map key-type="String" value-type="String"/>
             </field>
+            <field name="languages" persistence-modifier="persistent">
+                <collection type="String"/>
```
Comment (tzaeschke) [inline tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo:110]: Shouldn't that be `<collection element-type="String"/>`?

File: tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo [hunk 4]
```diff
@@ -241,6 +244,9 @@ has datastore identity.
             <property name="phoneNumbers" persistence-modifier="persistent">
                 <map key-type="String" value-type="String"/>
             </property>
+            <property name="languages" persistence-modifier="persistent">
+                <collection type="String"/>
```
Comment (tzaeschke) [inline tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo:248]: Shouldn't that be `<collection element-type="String"/>`?

File: tck/src/main/resources/sql/derby/applicationidentity/schema.sql [hunk 5]
```diff
@@ -407,6 +408,11 @@ CREATE TABLE employee_phoneno_type (
     TYPE VARCHAR(16) NOT NULL
 );
 
+CREATE TABLE employee_language (
```
Comment (tzaeschke) [inline tck/src/main/resources/sql/derby/applicationidentity/schema.sql:411]: Use plural? 'employee_languages`?

**Review by tzaeschke — APPROVED** [review]
LGTM % some minor questions/suggestions

File: tck/src/main/java/org/apache/jdo/tck/lifecycle/PMsCanSharePCClassesButNotPCInstances.java [hunk 6]
```diff
@@ -110,15 +110,15 @@ protected void insertObjects(boolean sharedPC) {
 
     while (!attemptsComplete()) {
       try {
-        Thread.sleep(100);
+        Thread.sleep(200);
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/lifecycle/PMsCanSharePCClassesButNotPCInstances.java:113]: I had issues with the test case. I changed back the value to 100.

File: tck/src/main/java/org/apache/jdo/tck/pc/company/Person.java [hunk 7]
```diff
@@ -261,6 +265,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The set of languages for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this.languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/company/Person.java:283]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedFC/FCAppPerson.java [hunk 8]
```diff
@@ -301,6 +309,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The map of phoneNumbers for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this.languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedFC/FCAppPerson.java:327]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedJPA/JPAAppPerson.java [hunk 9]
```diff
@@ -305,6 +315,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The map of phoneNumbers for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this.languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedJPA/JPAAppPerson.java:333]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedPC/PCAppPerson.java [hunk 10]
```diff
@@ -294,6 +298,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(_languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The set of languages for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this._languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedPC/PCAppPerson.java:316]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedFC/FCDSPerson.java [hunk 11]
```diff
@@ -297,6 +305,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The map of phoneNumbers for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this.languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedFC/FCDSPerson.java:323]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedPC/PCDSPerson.java [hunk 12]
```diff
@@ -294,6 +298,24 @@ public void setPhoneNumbers(Map<String, String> phoneNumbers) {
...
+    return Collections.unmodifiableSet(_languages);
+  }
+
+  /**
+   * Set the languages set to be in this person.
+   *
+   * @param languages The map of phoneNumbers for this person.
+   */
+  public void setLanguages(Set<String> languages) {
+    this._languages = languages;
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/pc/companyAnnotatedPC/PCDSPerson.java:316]: Good catch.

File: tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java [hunk 13]
```diff
@@ -3031,6 +3048,450 @@ public void testQuery19f() {
...
+  /**
+   * Navigation through multi-valued field.
+   *
+   * <p>This query selects all FullTimeEmployee instances from the candidate collection speaking
+   * German (i.e. the language set includes the string "German").
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  @Execution(ExecutionMode.CONCURRENT)
+  public void testQuery20f() {
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java:3176]: Test methods named xxe (e.g. testQuery20e) are used for a named query version of the JDO query. The JDOQLTyped version is using the name pattern xxf, see class javadoc of SampleReadQueries.

File: tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java [hunk 14]
```diff
@@ -3031,6 +3048,450 @@ public void testQuery19f() {
...
+  /**
+   * Navigation through multi-valued field.
+   *
+   * <p>This query selects all FullTimeEmployee instances from the candidate collection having a
+   * home phone number (i.e. the phoneNumbers map includes an entry with key "home").
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  @Execution(ExecutionMode.CONCURRENT)
+  public void testQuery21f() {
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java:3323]: See above.

File: tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java [hunk 15]
```diff
@@ -3031,6 +3048,450 @@ public void testQuery19f() {
...
+  /**
+   * Navigation through multi-valued field.
+   *
+   * <p>This query selects all FullTimeEmployee instances from the candidate collection having a
+   * phone number "1111" (i.e. the phoneNumbers map includes an entry with value "1111").
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  @Execution(ExecutionMode.CONCURRENT)
+  public void testQuery22f() {
```
[the remark this answers is not in the document]
Comment (mboapache) [reply tck/src/main/java/org/apache/jdo/tck/query/api/SampleReadQueries.java:3473]: See above.

File: tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo (hunk 1 above)
Comment (mboapache) [reply tck/src/main/resources/jdo/applicationidentity/org/apache/jdo/tck/pc/company/package.jdo:135]: Good catch!

File: tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo (hunk 3 above)
Comment (mboapache) [reply tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo:110]: Good catch!

File: tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo (hunk 4 above)
Comment (mboapache) [reply tck/src/main/resources/jdo/datastoreidentity/org/apache/jdo/tck/pc/company/package.jdo:248]: Good catch!

File: tck/src/main/resources/sql/derby/applicationidentity/schema.sql (hunk 5 above)
Comment (mboapache) [reply tck/src/main/resources/sql/derby/applicationidentity/schema.sql:411]: I renamed the join table to employee_languages

**Pull request merged**